lambdas/services/lloyd_george_validator.py

```diff
@@ -1,7 +1,44 @@
+import re
+
+from lambdas.models.nhs_document_reference import NHSDocumentReference
+
+
 class LGFileTypeException(ValueError):
-    """One or more of the files do not match the required file type."""
+    """One or more of the files do not match the required file type"""
     pass
 
-def validate_lg_file_type(file_type):
+class LGFileNameException(ValueError):
+    """One or more of the files do not match naming convention"""
+    pass
+
+class LGInvalidFilesException(Exception):
+    """One or more of the files are not valid"""
+    pass
+
+def validate_lg_file_type(file_type: str):
     if file_type != 'application/pdf':
         raise LGFileTypeException
+
+def validate_file_name(name: str):
+    lg_regex = r'[0-9]+of[0-9]+_Lloyd_George_Record_\[[A-Za-z ]+]_\[[0-9]{10}]_\[\d\d-\d\d-\d\d\d\d].pdf'
+    if not re.fullmatch(lg_regex, name):
+        raise LGFileNameException
+
+def check_for_duplicate_files(file_list: list[str]):
+    if len(file_list) > len(set(file_list)):
+        raise LGInvalidFilesException
```

Comment:
Instead of 3 seperate exceptions for this, would it not be better to make a single LGValidationException and allow it to be raised with a message inside detailing the specific failure which could be used to inform the user as noted above?

Reply:
I've changed it to one